examples/playbooks/role_vars_prefix_detection.yml

```diff
@@ -0,0 +1,51 @@
+---
+- name: Test role-prefix
+  hosts: localhost
+  connection: local
+  roles:
+    - role_vars_prefix_detection
+
+    - role: role_vars_prefix_detection
+      var1: val1
+
+    - role: role_vars_prefix_detection
+      var1: val1
+      become: true
+      vars:
+        var2: val2
+
+    - role: role_vars_prefix_detection
+      become: true
+      environment:
+        FOO: /bar/barr
+      role_vars_prefix_detection_var1: val1
+
+    - role: role_vars_prefix_detection
+      vars:
```

Comment:
Side blurb, I would think one would want to put role variables under the `vars` key instead of mixing them in the same dictionary as the role name. This would prevent confusion with a playbook's role keywords.